netbox/utilities/middleware.py

```diff
@@ -19,9 +19,10 @@ def __init__(self, get_response):
     def __call__(self, request):
         if LOGIN_REQUIRED and not request.user.is_authenticated:
             # Redirect unauthenticated requests to the login page. API requests are exempt from redirection as the API
-            # performs its own authentication.
+            # performs its own authentication. Also metrics can be read without login.
             api_path = reverse('api-root')
-            if not request.path_info.startswith(api_path) and request.path_info != settings.LOGIN_URL:
+            if (not (request.path_info.startswith(api_path) or request.path_info.startswith('/metrics'))
```

Comment:
`startswith()` will actually support passing a tuple of elements to test against, which should allow this to all be on one time.

```suggestion
            if not request.path_info.startswith((api_path, '/metrics')) and request.path_info != settings.LOGIN_URL:
```

Reply:
That's even more elegant. Thanks for this change!

Reply:
This should be covered now, although I somehow managed to push another version at the same time, which unfortunately outdated this change for the simple merge.